src/main/java/org/spongepowered/api/item/inventory/query/QueryType.java

```diff
@@ -29,21 +29,52 @@
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
 /**
- * Represents a possible type of operation for an {@linkplain
- * Inventory#query(QueryOperation...) inventory query}.
- *
- * @param <T> The argument type for the query
+ * Represents a possible type of operation for an {@link Inventory#query inventory query}.
  */
-@CatalogedBy(QueryOperationTypes.class)
-public interface QueryOperationType<T> extends CatalogType {
+@CatalogedBy(QueryTypes.class)
+public interface QueryType extends CatalogType {
+
+    /**
+     * A type of query that requires no parameters. It can directly be used as a query.
+     */
+    interface NoParam extends QueryType, Query {
```

Comment:
`NoParam` shouldn't directly extend `Query`; should rather add a factory method and leave this as an impl detail

Reply:
Why not? They're basically just predefined Queries (previously InventoryTransformations)
I don't see a need for a factory method creating more instances.

`inv.query(QueryTypes.REVERSE)`
`playerInv.query(QueryTypes.PLAYER_PRIMARY_HOTBAR_FIRST)`

Reply:
I don't see a need for object types to implement `execute` method.